src/rickbox/app.py

```diff
@@ -8,5 +10,47 @@ def hello_world():
     return 'Hello, World!'
 
 
+@app.route('/users/by_username/<username>')
```

Comment:
I don't like how this is structured. It doesn't seem clean or RESTful. I'm thinking using query parameters might be better. Something like `/users?username=jason` or `/users?id=12345`. It could also be changed to not care about the type. It could take in the value and then the query could check username and id. Something like `select * from users where id=%s or username=%s`. We could reduce this to one endpoint and have the path be much simpler.

Reply:
I mostly agree! This was a test endpoint that I let slip in and totally my bad for not proofreading better.

The part I slightly disagree on is using a query param. I feel like we might lose marks for not making this as RESTful as possible. Maybe we can choose either IDs or usernames, then just not support the one we don't choose? `/users/<username>` or `/users/<user_id>`. It seems to be convention to support `/users/<username>` since usernames provide better UX (easier to remember). 

If we want to support both, yeah query params seem like the way to go.

Reply:
Tried to do this change and test it... apparently I am banned from SSHing into the school for some reason.

```
➜  ssh adupuis2@fcslinux.cs.unb.ca
ssh: connect to host fcslinux.cs.unb.ca port 22: Connection refused
```

I emailed Sean Seeley about it. In the meantime, I don't want to push any changes without testing on info3103.cs.unb.ca first. I'll head into school early tomorrow and work on it before Stats.

Sorry I didn't get this in today like I said I would.